WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsButtonViewModel.swift

```diff
@@ -0,0 +1,41 @@
+import Foundation
+import Combine
+import UIKit
+import SwiftUI
+
+final class NotificationsButtonViewModel: ObservableObject {
+    @Published private(set) var counter = 0
+    @Published private(set) var image: UIImage?
+
+    private var cancellables: [AnyCancellable] = []
+
+    init() {
+        refresh()
+
+        NotificationCenter.default
+            .publisher(for: NSNotification.ZendeskPushNotificationReceivedNotification)
+            .sink { [weak self] _ in self?.refresh() }
+            .store(in: &cancellables)
+
+        NotificationCenter.default
+            .publisher(for: NSNotification.ZendeskPushNotificationClearedNotification)
+            .sink { [weak self] _ in self?.refresh() }
+            .store(in: &cancellables)
+
+        UIApplication.shared
+            .publisher(for: \.applicationIconBadgeNumber)
+            .sink { [weak self] _ in self?.refresh() }
+            .store(in: &cancellables)
+    }
+
+    private func refresh() {
+        counter = UIApplication.shared.applicationIconBadgeNumber - ZendeskUtils.unreadNotificationsCount
+
+        if counter > 0 {
+            image = UIImage(systemName: "bell.badge")?
+                .withConfiguration(UIImage.SymbolConfiguration(paletteColors: [.systemRed, .label]))
+        } else {
+            image = UIImage(systemName: "bell")
+        }
+    }
+}
```

Comment:
What do you think changing this class to a function? Something like:

```swift
func notificationIcon() -> AnyPublisher<UIImage?, Never> {
    Publishers.MergeMany(
        NotificationCenter.default
            .publisher(for: NSNotification.ZendeskPushNotificationReceivedNotification)
            .map { _ in }
            .eraseToAnyPublisher(),

        NotificationCenter.default
            .publisher(for: NSNotification.ZendeskPushNotificationClearedNotification)
            .map { _ in }
            .eraseToAnyPublisher(),

        UIApplication.shared
            .publisher(for: \.applicationIconBadgeNumber).map { _ in }
            .eraseToAnyPublisher()
    )
    .map {
        let counter = UIApplication.shared.applicationIconBadgeNumber - ZendeskUtils.unreadNotificationsCount

        if counter > 0 {
            return UIImage(systemName: "bell.badge")?
                .withConfiguration(UIImage.SymbolConfiguration(paletteColors: [.systemRed, .label]))
        } else {
            return UIImage(systemName: "bell")
        }
    }
    .eraseToAnyPublisher()
}
```

Reply:
A ViewModel is a bit more versatile, easier to debug, and a bit easier to use. I've done observable chains in the past and I tend to avoid them now.